File: old_runs/triangle-c_run/scenario_slow_perf_instructions_2/necessary/triangle.c
#include "triangle.h"

void delay() {
  const struct timespec ms = {0, 0.001*1e9}; //tv_sec=0, tv_nsec (0.001 seconds)
  nanosleep(&ms,NULL); /*ignores possible errors*/
}

int classify_triangle(double a, double b, double c) {
  double tmp;

  if(b > c) {
    tmp = b;
    b = c;
    c = tmp;
  }

  // Sort the sides so that a <= b <= c
  tmp = b;

  if(a > c) {
    tmp = a;
    a = c;
    c = tmp;
  }

  if(b > c) {
    tmp = b;
    b = c;
    c = tmp;
  }

  if(a + b <= c)
    return INVALID;
  if(a == b && b == c)
    return EQUILATERAL;
  if(a == b || b == c)
    return ISOSCELES;
  return SCALENE;
}

File: final/triangle-py-600/scenario_slow_perf_cache_misses_3/necessary/triangle.py

import enum
def delay():
    time.sleep(0.001)
import time

class TriangleType(enum.Enum):
    INVALID, EQUILATERAL, ISOSCELES, SCALENE = (0, 1, 2, 3)



def classify_triangle(a, b, c):
    
    if a > b:
        tmp = a
        a = b
        b = tmp
    
    if b > c:
        tmp = b
        b = c
        c = tmp
    if a + b <= c:
        return TriangleType.INVALID
    else:
        if a == b and b == c:
            INVALID, EQUILATERAL, ISOSCELES, SCALENE = (0, 1, 2, 3)
            return TriangleType.EQUILATERAL
        else:
            if a == b or b == c:
                return TriangleType.ISOSCELES
            else:
                return TriangleType.SCALENE
